LocationManager/INTULocationManager/INTULocationManager.m

```diff
@@ -692,6 +758,110 @@ - (INTULocationAccuracy)achievedAccuracyForLocation:(CLLocation *)location
     }
 }
 
+#pragma mark Internal heading methods
+
+/**
+ Adds the given heading request to the array of requests and starts heading updates.
+ */
+- (void)addHeadingRequest:(INTUHeadingRequest *)headingRequest
+{
+    NSAssert(headingRequest, @"Must pass in a non-nil heading request.");
+
+    // If heading services are not available, just return
+    if ([INTULocationManager headingServicesState] == INTUHeadingServicesStateUnavailable) {
+        if (headingRequest.block) {
+            headingRequest.block(nil, INTUHeadingStatusUnavailable);
+        }
+        INTULMLog(@"Heading Request (ID %ld) NOT added since device heading is unavailable.", (long)headingRequest.requestID);
+        return;
+    }
+
+    __INTU_GENERICS(NSMutableArray, INTUHeadingRequest *) *newHeadingRequests = [NSMutableArray arrayWithArray:self.headingRequests];
+    [newHeadingRequests addObject:headingRequest];
+    self.headingRequests = newHeadingRequests;
+    INTULMLog(@"Heading Request added with ID: %ld", (long)headingRequest.requestID);
+
+    if (!self.isUpdatingHeading) {
+        self.isUpdatingHeading = YES;
```

Comment:
Can you update this little section of code where we start heading updates to match the existing code that starts location updates? Specifically, start the updates first, then check if the flag was NO and if so log a message that updates started, and finally set the flag to YES. We should maintain parallel logic unless there is a reason not to.